### PR TITLE
Support handling actions with all payment components

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Add generic action handling to all payment components.

## Checklist 
- [x] Changes are tested manually

COAND-650

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/component/BacsDirectDebitDialogFragment.kt [hunk 1]
```diff
@@ -78,7 +79,8 @@ internal class BacsDirectDebitDialogFragment : BaseComponentDialogFragment() {
     }
 
     override fun onBackPressed(): Boolean {
-        return if (bacsDirectDebitComponent.delegate.handleBackPress()) {
+        val delegate = bacsDirectDebitComponent.delegate as? BacsDirectDebitDelegate
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/component/BacsDirectDebitDialogFragment.kt:82]: You'll be able to call `component.handleBackPress()` directly after [this commit](https://github.com/Adyen/adyen-android/pull/961/commits/1bdbdc8686a14561377eea9caf78b19bf06ee3ef)

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt [hunk 2]
```diff
@@ -65,18 +67,29 @@ class BcmcComponentProvider(
...
+                paymentMethod = paymentMethod,
+                publicKeyRepository = publicKeyRepository,
+                componentParams = componentParams,
+                cardValidationMapper = cardValidationMapper,
+                cardEncrypter = cardEncrypter,
+                analyticsRepository = analyticsRepository,
+                submitHandler = SubmitHandler()
+            )
+
+            val genericActionDelegate = GenericActionComponent.PROVIDER.getDelegate(
```
Comment (jreij) [inline bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt:83]: I think we should use `GenericActionComponentProvider(overrideComponentParams)` here.

My suspicion is that if we launch a component from drop-in, and set a custom config for it, the drop-in config will override the custom config for the component delegate but not for the action delegate. Can you make a quick test to see if this true?

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/component/BacsDirectDebitDialogFragment.kt (hunk 1 above)
Comment (OscarSpruit) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/component/BacsDirectDebitDialogFragment.kt:82]: Oh nice! Didn't see that

**Review by jreij — APPROVED** [review]

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt (hunk 2 above)
Comment (OscarSpruit) [reply bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt:83]: You are right, let me fix it

**Review by jreij — APPROVED** [review]

**Pull request merged**